### PR TITLE
Fixed object-literal-sort-keys rule ignores quoted keys #1512

### PR DESCRIPTION
**Pull request by bolatovumar**

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/tslint](https://github.com/palantir/tslint), @bolatovumar! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: src/rules/objectLiteralSortKeysRule.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ class ObjectLiteralSortKeysWalker extends Lint.RuleWalker {
         if (sortedState) {
             const lastSortedKey = this.lastSortedKeyStack[this.lastSortedKeyStack.length - 1];
             const keyNode = node.name;
-            if (keyNode.kind === ts.SyntaxKind.Identifier) {
+            if (keyNode.kind === ts.SyntaxKind.Identifier || keyNode.kind === ts.SyntaxKind.StringLiteral) {
                 const key = (<ts.Identifier> keyNode).text;
```
Comment (adidahiya) [inline src/rules/objectLiteralSortKeysRule.ts]: this type assertion is no longer accurate with the change in the line above. I suggest refactoring into a type guard

```ts
function isIdentifierOrStringLiteral(node: ts.Node): node is (ts.Identifier | ts.StringLiteral) {
    return node.kind === ts.SyntaxKind.Identifier || node.kind === ts.SyntaxKind.StringLiteral;
}
```

File: src/language/walker/syntaxWalker.ts [hunk 2]
```diff
@@ -178,6 +178,10 @@ export class SyntaxWalker {
         this.walkChildren(node);
     }
 
+    protected visitJsxSpreadAttribute(node: ts.JsxSpreadAttribute) {
```
Comment (adidahiya) [inline src/language/walker/syntaxWalker.ts]: this change is unrelated; please submit this in a separate PR (we squash commits before merging so it would be confusing to find this in the bugfix commit)

Comment (bolatovumar) [inline src/language/walker/syntaxWalker.ts]: My bad. This accidentally ended up here.

File: src/rules/objectLiteralSortKeysRule.ts (hunk 1 above)
Comment (bolatovumar) [inline src/rules/objectLiteralSortKeysRule.ts]: Will fix.

**Comment by bolatovumar** [comment]
@adidahiya Fixing this issue breaks the build as many files in the source code have quoted keys which are not in alphabetical order. What would the ideal resolution be here? Should I position them alphabetically and create a separate pull request? Or submit everything as one pull request? Something else entirely?

**Comment by JKillian** [comment]
Doing it as a separate PR sounds good to me 👍

**Comment by bolatovumar** [comment]
Issue a new pull request [here](https://github.com/palantir/tslint/pull/1529)
